operator-common/src/main/java/io/strimzi/operator/common/model/PemTrustSet.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.model;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.operator.common.Util;
+
+import java.nio.charset.StandardCharsets;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Class to represent the set of trusted certificates used by TLS clients
+ */
+public class PemTrustSet {
+    private static final String CERT_SUFFIX = ".crt";
+
+    private final Set<X509Certificate> trustedCertificates;
+    private final Map<String, byte[]> trustedCertificateMap;
+    private String secretName;
+    private String secretNamespace;
+
+    /**
+     * Constructs the PemTrustSet. This also validates the provided Secret contains valid certificates.
+     * @param secret Kubernetes Secret containing the trusted certificates
+     */
+    public PemTrustSet(Secret secret) {
+        if (secret == null || secret.getData() == null) {
+            throw new RuntimeException("Cannot extract trust set from null secret");
+        }
+        Optional.of(secret)
+                .map(Secret::getMetadata)
+                .ifPresent(objectMeta -> {
+                    secretName = objectMeta.getName();
+                    secretNamespace = objectMeta.getNamespace();
+                });
+        trustedCertificateMap = extractCerts(secret);
+        trustedCertificates = new HashSet<>(validateCertificates().values());
+    }
+
+    /**
+     * @return The set of trusted certificates as X509Certificate objects
+     */
+    public Set<X509Certificate> trustedCertificates() {
+        return trustedCertificates;
+    }
+
+    /**
+     * @return The set of trusted certificates as byte arrays
+     */
+    public Set<byte[]> trustedCertificatesBytes() {
+        return new HashSet<>(trustedCertificateMap.values());
+    }
+
+    /**
+     * @return The set of trusted certificates as a concatenated String
+     */
+    public String trustedCertificatesString() {
+        return trustedCertificateMap.values()
+                .stream()
+                .map(bytes -> new String(bytes, StandardCharsets.US_ASCII))
+                .collect(Collectors.joining("\n"));
+    }
+
+    /**
+     * Validate the set of certificates in this PemTrustSet.
```

Comment:
This method returns something, which you're not documenting. So this might be better named something like `asX509Certificates()`.